### DISCUSSION TITLE
Property driven text-size

### DISCUSSION THREAD

**Discussion by JonFreer** [maplibre/maplibre-gl-js#4948] [Q&A]
Hi all,

I am trying to render text at a fixed geographical scale. I have been able to achieve this for geometry properties such as line-width by interpolating based on a zoom value and some property value.  However, when I try this approach with text-size it fails to work. It does work when just interpolating based on a fixed constant and zoom value. I have prepared a jsbin with an example of this.

Constant:
```
        'text-size': [
                        "interpolate",
                        ["exponential", 2],
                        ["zoom"],
                        2,
                        ["*", 1000000, ["^", 2, -14]],
                        24,
                        ["*", 1000000, ["^", 2, 8]],
                    ],
```

Property Driven:
```
          'text-size': [
                        "interpolate",
                        ["exponential", 2],
                        ["zoom"],
                        2,
                        ["*", ['get', 'size'], ["^", 2, -14]],
                        24,
                        ["*", ['get', 'size'], ["^", 2, 8]],
                    ],

```
https://jsbin.com/ginahegazu/edit?html,output

Would appreciate any pointers on this.

Many thank,

Jon

**Comment by HarelM** [comment]
As can be read in the docs, text-size supports interpretation but is not data driven: 
https://maplibre.org/maplibre-style-spec/layers/#text-size

**Comment by zerda-ocm** [comment]
I have the same requirement and have a working solution with some minor limitations. Although property driven text-size is not officially supported, it seems to be implemented in general. However it has a major bug: it fails over a large zoom-range.

https://jsbin.com/xiyahatoli/1/edit?html,output

This fails:
```
'text-size': [
    "interpolate",
    ["exponential", 2],
    ["zoom"],
    2, ["*", ['get', 'size'], ["^", 2, -14]],
    24, ["*", ['get', 'size'], ["^", 2, 8]],
],
```

This will work with the limitation, than the text can't get infinite large.
```
'text-size': [
    "interpolate",
    ["exponential", 2],
    ["zoom"],
    0, ["*", ['get', 'size'], ["^", 2, -16]],
    4, ["*", ['get', 'size'], ["^", 2, -12]],
    8, ["*", ['get', 'size'], ["^", 2, -8]],
    12, ["*", ['get', 'size'], ["^", 2, -4]],
    16, ["*", ['get', 'size'], ["^", 2, 0]],
    20, ["*", ['get', 'size'], ["^", 2, 4]],
    24, ["*", ['get', 'size'], ["^", 2, 8]],
],
```
